modules/lib/lib_gui.py

```diff
@@ -0,0 +1,79 @@
+import tkinter as tk
+import pyperclip
+from tkinter import messagebox
+from lib_pass_gen import PasswordGenerator
+
+# This class allows us to create entries with placeholders so it can provide more detail to use wihout additional labels.
+class EntryWithPlaceholder(tk.Entry):
+    def __init__(self, master=None, placeholder="PLACEHOLDER", textvariable=None, color="grey"):
+        super().__init__(master)
+
+        self.placeholder = placeholder
+        self.placeholder_color = color
+        self.default_fg_color = self["fg"]
+
+        if textvariable:
+            self.configure(textvariable=textvariable)
+
+        self.bind("<FocusIn>", self.foc_in)
+        self.bind("<FocusOut>", self.foc_out)
+
+        self.put_placeholder()
+
+    def put_placeholder(self):
+        self.insert(0, self.placeholder)
+        self["fg"] = self.placeholder_color
+
+
+
+    def foc_in(self, *args):
```

Comment:
What is `foc`. I think it's `focus` from the context but there is really no reason to not call it `focus_in`. Tt's really not that long

Reply:
good idea. It'll get more readable